From 88c08e7e098d2962e1bdbeb56a48d4a20f73f0ae Mon Sep 17 00:00:00 2001
From: Jelle van der Waa <jelle@archlinux.org>
Date: Fri, 23 Sep 2022 14:35:20 +0200
Subject: [PATCH] Support file 5.43

File 5.43 changed it's output format for pyc files to something new:

Byte-compiled Python module for CPython 3.9, timestamp-based, .py
timestamp: Sun Sep  5 18:47:25 2021 UTC, .py size: 14217 bytes
---
 diffoscope/comparators/python.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/diffoscope/comparators/python.py b/diffoscope/comparators/python.py
index 5cf30c7d..1b9d4220 100644
--- a/diffoscope/comparators/python.py
+++ b/diffoscope/comparators/python.py
@@ -36,7 +36,7 @@ re_memory_address = re.compile(r" at 0x\w+(?=, )")
 
 class PycFile(File):
     DESCRIPTION = "Python .pyc files"
-    FILE_TYPE_RE = re.compile(r"^python .*byte-compiled$")
+    FILE_TYPE_RE = re.compile(r"(^python .*byte-compiled$|^Byte-compiled Python module for CPython .*)")
     FALLBACK_FILE_EXTENSION_SUFFIX = {".pyc"}
 
     def compare_details(self, other, source=None):
-- 
2.37.3

